From 750f5f97fd4d630987e9d4ce32073aad845b3c23 Mon Sep 17 00:00:00 2001
From: Jason van Zyl <jason@tesla.io>
Date: Tue, 20 May 2014 20:26:27 -0400
Subject: [PATCH 6/6] [MNG-5613] fix NullPointerException when error occur
 during project dependency graph creation Submitted by: Sebastian Le Merdy
 <sebastian.lemerdy@gmail.com> From: https://github.com/apache/maven/pull/13

---
 maven-core/src/main/java/org/apache/maven/DefaultMaven.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/maven-core/src/main/java/org/apache/maven/DefaultMaven.java b/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
index 6328819..ab47efd 100644
--- a/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
+++ b/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
@@ -267,13 +267,13 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request )
         //
         ProjectDependencyGraph projectDependencyGraph = createProjectDependencyGraph( projects, request, result, true );
 
-        session.setProjects( projectDependencyGraph.getSortedProjects() );
-        
         if ( result.hasExceptions() )
         {
             return result;
         }
 
+        session.setProjects( projectDependencyGraph.getSortedProjects() );
+
         try
         {
             session.setProjectMap( getProjectMap( session.getProjects() ) );
-- 
1.9.0

